POEApi.Transport/HttpTransport.cs

```diff
@@ -106,6 +126,43 @@ public bool Authenticate(string email, SecureString password, bool useSessionID)
             return true;
         }
 
+        private void CloudFlareSessionIdLogin()
+        {
+            try
+            {
+                using (var clearanceHandler = new ClearanceHandler {MaxRetries = 2})
+                using (var handler = new HttpClientHandler() {CookieContainer = credentialCookies, Proxy = getProxySettings()})
+                {
+                    clearanceHandler.InnerHandler = handler;
+
+                    using (var client = new HttpClient(clearanceHandler))
+                    {
+                        var result = client.GetStringAsync(new Uri(loginURL)).Result;
+                    }
+                }
+            }
+            catch (AggregateException ex) when (ex.InnerException is CloudFlareClearanceException)
+            {
+                // After all retries, clearance still failed.
+                throw new Exception("Cloud flare clearance failed, please wait one minute and try again");

```

Comment:
Include the `AggregateException ex` as the second parameter to the `Exception` constructor, so it and its details are not lost.

Reply:
Great spot!